src/components/cartdetail/PageCartDetail.vue

```diff
@@ -50,15 +53,59 @@ import gql from 'graphql-tag';
 import cartMixin from '@/mixins/cartMixin';
 import CartContent from '@/components/cartdetail/CartContent.vue';
 import PriceCalculation from '@/components/common/PriceCalculation.vue';
+import DiscountCode from '@/components/common/DiscountCode.vue';
 import DisplayableMoneyFragment from '@/components/DisplayableMoney.gql';
 
 export default {
-  components: { CartContent, PriceCalculation },
+  components: {
+    CartContent,
+    PriceCalculation,
+    DiscountCode,
+  },
 
   data: () => ({
     me: null,
+    buttonState: null,
+    serverError: null,
   }),
 
+  methods: {
+    addDiscountCode(code) {
+      return this.$apollo.mutate({
+        mutation: gql`
+          mutation AddDiscountCode($id: String!, $version: Long!, $code: String!) {
+            updateMyCart(id: $id, version: $version, actions: {
+              addDiscountCode: {
+                code: $code
+              }
+            }){
+              id
+              version
+            }
+          }`,
+        variables: {
+          id: this.me.activeCart.id,
+          version: this.me.activeCart.version,
+          code,
+        },
+      });
+    },
+
+    submitCode(code) {
+      this.serverError = null;
+      this.buttonState = 'loading';
+      this.addDiscountCode(code)
+        .then(() => {
+          this.buttonState = 'success';
+          this.$apollo.queries.me.refetch();
```

Comment:
This refetch will only work with queries that are called "me" by the developer (not to confuse with the "me" field). That makes it very likely for new developers to make mistakes with it, as they will write queries with other names and expect that when e.g. adding a discount code their view of the cart will be updated too, but it will not. This is the reason why we discarded this approach in previous updates.

Instead, take a look at the approach being used elsewhere (e.g. `productdetail/AddToCartForm.vue`). It updates everything automatically, just by looking at the type and ID of the resource. Only requirement is to have all the queried fields in the update response, but it's explained in the readme (https://github.com/commercetools/sunrise-spa#add-any-queried-fields-to-the-mutation) and in the related issue you can read that this problem will be fixed in Apollo Client v3.0. And that's why we decided that this was the best option to follow :)